### PR TITLE
Add Demos CI workflow

### PR DESCRIPTION
**Pull request by vconst**

### CODE REVIEW

**Review by nightskylark — DISMISSED** [review]

**Review by MikeVitik — DISMISSED** [review]

**Review by babich-a — DISMISSED** [review]

**Review by DokaRus — DISMISSED** [review]

File: .github/workflows/demos_tests.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        BASE_BRANCH=$(node -p -e "require('./package.json').version.slice(0, 4).replace('.', '_')")
+        test -d ./devextreme-demos || git clone -b $BASE_BRANCH https://github.com/devexpress/devextreme-demos ./devextreme-demos
+
+    # - name: Restore npm cache
+    #   uses: actions/cache@v2
+    #   with:
+    #     path: ./node_modules
+    #     key: ${{ runner.os }}-node-${{ hashFiles('**/package.json') }}
+    #     restore-keys: |
+    #       ${{ runner.os }}-node
```
Comment (alexserov) [inline .github/workflows/demos_tests.yml:33]:
```suggestion
    - name: Restore npm cache
      uses: actions/cache@v2
      with:
        path: ./node_modules
        key: ${{ runner.os }}-node-modules-${{ hashFiles('**/package-lock.json') }}
        restore-keys: ${{ runner.os }}-node-modules
```

File: .github/workflows/demos_tests.yml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+    - name: Run npm install
+      run: npm install --no-package-lock --no-audit --no-fund
```
Comment (alexserov) [inline .github/workflows/demos_tests.yml:36]:
```suggestion
      run: npm i --no-audit --no-fund
```

File: .github/workflows/demos_tests.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    - name: Build devextreme package
+      run: npm run build-npm-devextreme
+
+    - name: Pack devextreme package
+      working-directory: ./artifacts/npm/devextreme
+      run: npm pack
+
+    - name: Install packages for devextreme-demos
+      working-directory: ./devextreme-demos
+      run: npm install --no-package-lock --no-audit --no-fund
```
Comment (alexserov) [inline .github/workflows/demos_tests.yml:47]:
```suggestion
    - name: Restore npm cache for devextreme-demos
      working-directory: ./devextreme-demos
      uses: actions/cache@v2
      with:
        path: ./node_modules
        key: ${{ runner.os }}-node-modules-${{ hashFiles('**/package-lock.json') }}
        restore-keys: ${{ runner.os }}-node-modules

    - name: Install packages for devextreme-demos
      working-directory: ./devextreme-demos
      run: npm i --no-audit --no-fund
```

File: .github/workflows/demos_tests.yml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        CONSTEL: [jquery(1/4), jquery(2/4), jquery(3/4), jquery(4/4)]
+
+    runs-on: ubuntu-latest
+    name: ${{ matrix.CONSTEL }}
+    timeout-minutes: 20
+  
+    steps:
+    - name: Get sources
+      uses: actions/checkout@v2
+
```
Comment (alexserov) [inline .github/workflows/demos_tests.yml:77]: Do we need this step?
```suggestion
```

**Review by alexserov — APPROVED** [review]

**Pull request merged**